server/src/grpc/server.rs

```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
         grpc_debug!("(request_flight_release) request: {:?}", request);
         self.region.request_flight_release(request)
     }
+}
 
-    async fn request_waypoints(
-        &self,
-        request: Request<WaypointsRequest>,
-    ) -> Result<Response<WaypointsResponse>, Status> {
-        grpc_warn!(
-            "([{}] request_waypoints) compliance server.",
-            self.region.get_region()
-        );
-        grpc_debug!("(request_waypoints) request: {:?}", request);
-        self.region
-            .request_waypoints(self.waypoints.clone(), request)
+async fn update_waypoints(host: String, port: u16, waypoints: &HashMap<String, Coordinates>) {
+    let nodes: Vec<svc_gis_client_grpc::client::Waypoint> = waypoints
+        .iter()
+        .map(
+            |(label, coordinates)| svc_gis_client_grpc::client::Waypoint {
+                label: label.clone(),
+                location: Some(*coordinates),
+            },
+        )
+        .collect();
+
+    if nodes.is_empty() {
+        grpc_warn!("(waypoints_loop) no waypoints to update.");
+        return;
     }
 
-    async fn request_restrictions(
-        &self,
-        request: Request<RestrictionsRequest>,
-    ) -> Result<Response<RestrictionsResponse>, Status> {
-        grpc_warn!(
-            "([{}] request_restrictions) compliance server.",
-            self.region.get_region()
-        );
-        grpc_debug!("(request_restrictions) request: {:?}", request);
-        self.region
-            .request_restrictions(self.restrictions.clone(), request)
+    let request = tonic::Request::new(UpdateWaypointsRequest { waypoints: nodes });
+    let address = format!("{}:{}", host, port);
+    let result = GisClient::connect(address).await;
+    let Ok(mut client) = result else {
+        grpc_error!("Failed to connect to gRPC server: {}", result.unwrap_err());
+        return;
+    };
+
+    match client.update_waypoints(request).await {
+        Ok(response) => {
+            grpc_info!("RESPONSE={:?}", response);
```

Comment:
Should this really be an info message? Feels more like a debug message
Also add the function name prefix